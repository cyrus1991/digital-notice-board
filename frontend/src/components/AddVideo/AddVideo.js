import React from 'react';

class AddVideo extends React.Component {
  render() {
    return (
      <div className="AddVideo">
        <label className="edit-title">Title</label>
        <input type="text"/>
        <label className="edit-desc">Description</label>
        <input type="text"/>
        <label className="edit-date">Display Date</label>
        <input type="date"/>
        <label className="edit-date">Expiry Date</label>
        <input type="date"/>
        <label>YouTube Embeded Code</label>
        <textarea name="" id="" cols="30" rows="10"></textarea>
        <a href="#">Delete this video</a>
        <button type="submit">Add</button>
      </div>
    );
  }
}

export default AddVideo;