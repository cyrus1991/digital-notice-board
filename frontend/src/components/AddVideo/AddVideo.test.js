import React from 'react';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import AddVideo from './AddVideo';

Enzyme.configure({ adapter: new Adapter() });

// ------------------load Addvideo
// ---------------------------------
function loadAddVideo () {
    var app = shallow(<AddVideo />);
    const mainComponent = app.find('.AddVideo');
    return mainComponent;
}

// ------------------Matching with the snapshot
// ---------------------------------
it('matching the snapshot' , () => {
    const tree = renderer.create(<AddVideo />).toJSON()
    expect(tree).toMatchSnapshot()
})

// ------------------the 1st component
// ---------------------------------

describe('Title component', () => {

    it('The title Should be "Title" ', () => {
        const mainComponent = loadAddVideo ();
        const title = mainComponent.find('label').at(0)
        expect(title.text()).toBe('Title');
    });

    it('The input Should be text ', () => {
        const mainComponent = loadAddVideo ();
        const input = mainComponent.find('input').at(0)
        expect(input.prop('type')).toBe('text');
    });

})

// ------------------the 2nd component
// ---------------------------------

describe('Description component', () => {
    it('The title Should be "Description" ', () => {
        const mainComponent = loadAddVideo ();
        const title = mainComponent.find('label').at(1)
        expect(title.text()).toBe('Description');
    });

    it('The input Should be text ', () => {
        const mainComponent = loadAddVideo ();
        const input = mainComponent.find('input').at(1)
        expect(input.prop('type')).toBe('text');
    });
})

// ------------------the 3rd component
// ---------------------------------

describe('Display Date component', () => {
    it('The title Should be "Display Date" ', () => {
        const mainComponent = loadAddVideo ();
        const title = mainComponent.find('label').at(2)
        expect(title.text()).toBe('Display Date');
    });

    it('The input Should be date ', () => {
        const mainComponent = loadAddVideo ();
        const input = mainComponent.find('input').at(2)
        expect(input.prop('type')).toBe('date');
    });
})

// ------------------the 4th component
// ---------------------------------

describe('Expiry Date component', () => {
    it('The 4th title Should be "Expiry Date" ', () => {
        const mainComponent = loadAddVideo ();
        const title = mainComponent.find('label').at(3)
        expect(title.text()).toBe('Expiry Date');
    });

    it('The 4th input Should be date ', () => {
        const mainComponent = loadAddVideo ();
        const input = mainComponent.find('input').at(3)
        expect(input.prop('type')).toBe('date');
    });
})

// ------------------the 5th component
// ---------------------------------

describe('Youtube Embed Code component', () => {
    it('The 5th title Should be "Youtube Embed Code" ', () => {
        const mainComponent = loadAddVideo ();
        const title = mainComponent.find('label').at(4)
        expect(title.text()).toBe('Youtube Embed Code');
    });
})

// ------------------the 6th component
// ---------------------------------

describe('button component', () => {
    it('Check the click button', () => {
        const mainComponent = loadAddVideo ();
        const incrementBtn = mainComponent.find('button');
        incrementBtn.simulate('click');
    });
})