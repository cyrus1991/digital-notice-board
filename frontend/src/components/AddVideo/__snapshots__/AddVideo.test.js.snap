// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching the snapshot 1`] = `
<form
  className="AddVideo"
  onSubmit={undefined}
>
  <h1>
    Add Video
  </h1>
  <label>
    Title
  </label>
  <input
    className="form-control"
    id="title"
    type="text"
  />
  <label>
    Description
  </label>
  <input
    className="form-control"
    id="desc"
    type="text"
  />
  <label>
    Display Date
  </label>
  <input
    className="form-control"
    id="datestart"
    type="date"
  />
  <label>
    Expiry Date
  </label>
  <input
    className="form-control"
    id="dateend"
    type="date"
  />
  <label>
    Youtube Embed Code
  </label>
  <input
    className="form-control"
    id="url"
  />
  <a
    href="#"
  >
    Delete this Video
  </a>
  <br />
  <button
    className="btn btn-primary"
    type="submit"
  >
    Add
  </button>
</form>
`;
